src/Controls/src/Xaml/SimplifyOnPlatformVisitor.cs

```diff
@@ -0,0 +1,77 @@
+﻿#nullable disable
+
+namespace Microsoft.Maui.Controls.Xaml
+{
+	class SimplifyOnPlatformVisitor : IXamlNodeVisitor
+	{
+		public SimplifyOnPlatformVisitor(string targetFramework)
+		{
+			TargetFramework = targetFramework;
+		}
+
+		public string TargetFramework { get; }
+
+		public TreeVisitingMode VisitingMode => TreeVisitingMode.BottomUp;
+		public bool StopOnDataTemplate => false;
+		public bool VisitNodeOnDataTemplate => true;
+		public bool StopOnResourceDictionary => false;
+		public bool IsResourceDictionary(ElementNode node) => false;
+		public bool SkipChildren(INode node, INode parentNode) => false;
+
+		public void Visit(ValueNode node, INode parentNode)
+		{
+		}
+
+		public void Visit(MarkupNode node, INode parentNode)
+		{
+		}
+
+		public void Visit(ElementNode node, INode parentNode)
+		{
+			if (node.XmlType.Name != nameof(OnPlatformExtension) || node.XmlType.NamespaceUri != XamlParser.MauiUri)
+				return;
+			if (string.IsNullOrEmpty(TargetFramework))
+				return;
+
+			string target = null;
+			if (TargetFramework.Contains("-android"))
+				target = nameof(OnPlatformExtension.Android);
+			if (TargetFramework.Contains("-ios"))
+				target = nameof(OnPlatformExtension.iOS);
+			if (TargetFramework.Contains("-macos"))
+				target = nameof(OnPlatformExtension.macOS);
+			if (TargetFramework.Contains("-maccatalyst"))
```

Comment:
Is there no "Windows" platform extensions? Or do we not care in that case?

Reply:
This is kind of interesting since -windows could be wpf, or winforms or winappsdk in theory right?  We only support winappsdk now, but who knows in the future... So perhaps target framework is not enough to check for -windows at some point...

Reply:
there's no 1-1 mapping between .net TFM for windows and Maui targeted platform

Reply:
What are the "platform extensions" for windows? How does that switch happen at runtime?

Reply:
https://github.com/dotnet/maui/blob/main/src/Controls/src/Xaml/MarkupExtensions/OnPlatformExtension.cs#L13-L21 
the platform for the running app is set by the `IPlatformApplication` scaffolding the whole Maui thing

Reply:
https://github.com/dotnet/maui/pull/4491 added `WinUI` as a platform in the `OnPlatform` markup extension for windows... 

So, today we only have 1 windows target, perhaps we should optimize for the _now_.  If we need to, in the _future_ could we optimize such that at least `-windows` only has a limited number of options to check through and perhaps one `-windows` it's a slower path for this xamlc, but that's probably fine anyway because it's not android.

Reply:
this might be a good time to review our current platform list, and names, and what defines a platform. should iPadOS be it's own platform, what about deprecating windows in favour of winui, wpf, ...

Reply:
Am I reading this correctly, in that if it's not windows, then there's just no optimization in the code path like there is for the other known tfm's?  If that's the case I think this is fine as is and we should try and merge it in.

Reply:
yep, that's it, if the TFM is unlisted, we do nothing